app/controllers/api/admin/commodities_controller.rb

```diff
@@ -10,13 +10,17 @@ def show
       private
 
       def find_commodity
-        @commodity = Commodity.actual
-                              .declarable
-                              .by_code(params[:id])
-                              .take
+        @commodity = self.find_commodity_by_code(params[:id])
 
         raise Sequel::RecordNotFound if @commodity.goods_nomenclature_item_id.in? HiddenGoodsNomenclature.codes
       end
+
+      def self.find_commodity_by_code(code)
```

Comment:
@mcumcu can we do it as a model method rather than controller method?

Reply:
This class method has been moved to the Commodity model